Automattic/wp-calypso
# Add Clone Flow to Jetpack Cloud

## Proposed Changes

* Implements the clone flow into Jetpack Cloud

- [x] Implement Set Destination Screen
- [x] Implement Select Clone Period Screen
- [x] Implement Configure Screen
- [x] Detect Clone Progress
- [x] Implement Clone Progress Screen
### Known Issues
 * Switching from the Clone flow to Settings retains the state of 'updateRequestStatus' so can show the Finished screen.  Clicking the Finish Up button will return to normal with no ill effects.  The solution here is likely to have the state track a combination of siteId and role for these operations.  Since this isn't breaking and an edge case we're going to follow-up with a fix instead of making this PR larger.

## Testing Instructions



To test cloning you'll need two sites.  A source and a destination.  

The source site needs to have a Jetpack Backup plan and at least one backup.
The destination site needs FTP or SSH/SFTP access, and you'll need the credentials and server info handy.
You can utilize Jurassic Ninja to create both sites if needed.
You may want to make some changes so it'll be obvious the clone took place afterwards.

* Open the Jetpack Cloud Live link
* Select your source site
* Visit the VaultPress backup area
* Add `/clone` to the url - This is temporary until we add links
* Enter your server credentials and test them.  Once they're valid, click 'Finish Up'
* Select the Backup point you wish to clone
* Select the types of items, or leave the default of everything and Confirm
* Wait for the restore to finish and visit the destination site.

### UI Changes to existing pages
* Backup main page has had a header added with a Clone button
* Activity log main page has had the header changed, and includes a Clone button

## Pre-merge Checklist




## Screenshots for Translation Context
<img width="771" alt="Screenshot 2023-04-03 at 10 02 37 AM" src="https://user-images.githubusercontent.com/1992658/229534378-bbb1b968-7d2f-44ac-9659-013bf3dd2b47.png">
<img width="750" alt="Screenshot 2023-04-03 at 10 02 46 AM" src="https://user-images.githubusercontent.com/1992658/229534456-fb0eee3b-75e4-4a2b-9e43-d100656501c2.png">
<img width="726" alt="Screenshot 2023-04-03 at 10 05 53 AM" src="https://user-images.githubusercontent.com/1992658/229534474-bbe12d7d-3833-4d76-a67c-bc09fcdec020.png">
<img width="750" alt="Screenshot 2023-04-03 at 10 03 23 AM" src="https://user-images.githubusercontent.com/1992658/229534556-2dbe71e0-88ab-407c-8a4c-c7d8668fdbe8.png">
<img width="770" alt="Screenshot 2023-04-03 at 10 07 01 AM" src="https://user-images.githubusercontent.com/1992658/229534702-68b42d59-0596-4dd2-8a58-418d6711126b.png">
<img width="895" alt="Screenshot 2023-04-03 at 10 03 37 AM" src="https://user-images.githubusercontent.com/1992658/229535072-e07a8571-384e-41ba-a3e4-bb35b89ad2fc.png">
<img width="774" alt="Screenshot 2023-04-03 at 10 03 43 AM" src="https://user-images.githubusercontent.com/1992658/229534742-6a740ca3-9d62-43b3-86c3-f8280ba9861f.png">
<img width="787" alt="Screenshot 2023-04-03 at 10 05 29 AM" src="https://user-images.githubusercontent.com/1992658/229534779-fe1549e8-b306-4c8b-97e3-67c47e2e9f50.png">
